File: cpp/1310.xor-queries-of-a-subarray.cpp
#include <vector>
#include <algorithm>
using std::vector;

class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        int n = arr.size(), m = queries.size();
        vector<int> prefix(n);

        std::partial_sum(arr.begin(), arr.end(), prefix.begin(), [](int a, int b) -> int {
            return a ^ b;
        });

        vector<int> res(m, 0);

        for (int i = 0; i < m; i++) {
            res[i] = prefix[queries[i][0]] ^ arr[queries[i][0]] ^prefix[queries[i][1]];
        }

        return res;
    }
};

File: golang/1310.xor-queries-of-a-subarray.go
func xorQueries(arr []int, queries [][]int) []int {
	n, m := len(arr), len(queries)
	prefix := make([]int, n+1)
	res := make([]int, m)

	for i := 1; i <= n; i++ {
		prefix[i] = prefix[i-1] ^ arr[i-1]
	}

	for j := 0; j < m; j++ {
		u, v := queries[j][0], queries[j][1]+1
		res[j] = prefix[u] ^ prefix[v]
	}

	return res
}
